### DISCUSSION TITLE
DOM "jump" when animating the "x-transition:enter" in v3 versus "x-show.transition.in" in v2

### DISCUSSION THREAD

**Discussion by aaronbushnell** [alpinejs/alpine#1561] [7. Bugs (archived)] [answered]
I've noticed in Alpine v3 that the transition effect for "enter" behaves _slightly_ different than how it behaved in v2's "in" usage. Here's an reduced use-case:

## Using `x-show.transition.in` in a V2 component ([Codepen example](https://codepen.io/aaronbushnell/pen/YzZgJVO))
```html
<main x-data="{ 'active': true }">
	<button @click="active = ! active">Introduction</button>

	<div x-show.transition.in.duration.500ms="active === true">
		🧔🏼‍♂️ Hello, there.
	</div>

	<div x-show="active === false" x-cloak>
		🤖 General Kenobi!
	</div>
</main>
```
And here's the behavior that occurs. When the first state disappears it's instant and then it re-enters with the transition effect.
![v2](https://user-images.githubusercontent.com/315202/122601097-ca914580-d03e-11eb-87e7-6263ee0b5543.gif)

## Using `x-transition:enter` in a V3 component ([Codepen example](https://codepen.io/aaronbushnell/pen/mdWozqv))
```html
<main x-data="{ 'active': true }">
	<button @click="active = ! active">Introduction</button>

	<div x-show="active === true" x-transition:enter.duration.500ms>
		🧔🏼‍♂️ Hello, there.
	</div>

	<div x-show="active === false" x-cloak>
		🤖 General Kenobi!
	</div>
</main>
```
And here's the behavior that occurs. I slowed this to 0.5x speed so the behavior is more noticeable. The difference here is that the first `<div>` is immediately pushed downward by the second `<div>` creating a bit of a jarring effect.
![v3](https://user-images.githubusercontent.com/315202/122601572-7d61a380-d03f-11eb-86db-731174c79110.gif)

However, this could be intentional with the v3 changes and something I need to change on my end! Thanks so much for all the work you put into Alpine! 

**Update**: Appears to be related to a similar discussion #1229

**Comment by calebporzio** [comment, marked as answer]
Thanks for putting together such a nice demo. I've fixed this in Alpine - will be fixed in the next release.

**Reply by aaronbushnell** [reply]
Of course! Thanks so much for the work you pour into Alpine!

**Reply by BeetrootShoulders** [reply]
I'm getting this exact same issue even with the latest version:

https://jsfiddle.net/AthenasOwl/5m8xzt6h/2/

Any idea?

**Reply by ekwoka** [reply]
I'd say this is expected behavior.

You have your code set up to show both of those elements at the same time.

You can avoid this by having your CSS stack the elements on top of each other, like with a grid with 1 row and 1 column around them.

I don't think there is actually a good way for Alpine to do this internally, since it would have to do a lot of work to figure out what x-show might be related to each other and which not. Meanwhile, it's easy to do in userland.
